channeldb/invoices.go

```diff
@@ -312,16 +323,14 @@ func validateInvoice(i *Invoice) error {
 		return fmt.Errorf("max length a memo is %v, and invoice "+
 			"of length %v was provided", MaxMemoSize, len(i.Memo))
 	}
-	if len(i.Receipt) > MaxReceiptSize {
-		return fmt.Errorf("max length a receipt is %v, and invoice "+
-			"of length %v was provided", MaxReceiptSize,
-			len(i.Receipt))
-	}
 	if len(i.PaymentRequest) > MaxPaymentRequestSize {
 		return fmt.Errorf("max length of payment request is %v, length "+
 			"provided was %v", MaxPaymentRequestSize,
 			len(i.PaymentRequest))
 	}
+	if i.Terms.Features == nil {
+		return errors.New("invoice must have a feature vector")
+	}
```

Comment:
Can the addition of features be separated from the db changes?

Reply:
in what sense? another pr?

Reply:
i thought separate commit. isolate addition of the feature from moving to tlv. but nvm, it doesn't really matter